_test/test/build_integration_test.dart

```diff
@@ -92,5 +93,23 @@ void main() {
       await runBuild(trailingArgs: ['--output', 'build']);
       expect(dartSource.existsSync(), true);
     });
+
+    test('Re-snapshots if there is no asset graph', () async {
+      var assetGraph = assetGraphPathFor(p.url
+          .join('.dart_tool', 'build', 'entrypoint', 'build.dart.snapshot'));
+      await File(assetGraph).delete();
```

Comment:
This looks non-hermetic....

Reply:
We run a build in the setup so this should not be an issue.